context_propagation/starter.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+require 'logger'
+require 'temporalio/client'
+require_relative 'interceptor'
+require_relative 'say_hello_workflow'
+
+# Create a Temporal client
+client = Temporalio::Client.connect(
+  'localhost:7233',
+  'default',
+  logger: Logger.new($stdout, level: Logger::INFO),
+  # Add the context propagation interceptor to propagate the :my_user
+  # thread/fiber local
+  interceptors: [ContextPropagation::Interceptor.new(:my_user)]
+)
+
+# Set user as "Alice" which will get propagated in a distributed way through
+# the workflow and activity via Temporal headers
+Thread.current[:my_user] = 'Alice'
+
+# Start workflow, send signal, wait for completion, issue query
+puts 'Executing workflow with user "Alice"'
+handle = client.start_workflow(
+  ContextPropagation::SayHelloWorkflow,
+  'Bob',
```

Comment:
Are we enforcing 0- or 1-arity in Ruby as we wish we had done for others?

Reply:
No, and we didn't _wish_ to do this for the others, we just discourage multi-arity. Every SDK needs to support multiple parameters, even if we encourage a single one.

Reply:
Not sure about that